### PR TITLE
Support GCS

### LINKED ISSUE #3904: Support GCP for bundle storage

### PR DESCRIPTION
**Pull request by epicfaace**
Fixes #3904. Depends on https://github.com/codalab/codalab-worksheets/pull/3828 to be merged first.

### CODE REVIEW

File: codalab/lib/bundle_store.py [hunk 1]
```diff
@@ -422,5 +426,8 @@ def get_bundle_location(self, uuid):
         if storage_type == StorageType.AZURE_BLOB_STORAGE.value:
             file_name = "contents.tar.gz" if is_dir else "contents.gz"
             return f"azfs://{self._azure_blob_account_name}/bundles/{uuid}/{file_name}"
+        # if storage_type == StorageType.GCS_STORAGE.value:
+        #     file_name = "contents.tar.gz" if is_dir else "contents.gz"
```
Comment (epicfaace) [inline codalab/lib/bundle_store.py]: This is blocked on getting bundle locations / bundle store methods working, as that's the way we'll distinguish between Azure Blob and GCS in this method.

File: alembic/versions/2021121222_rm_storage_type_enum_a325a0756797.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+revision = 'a325a0756797'
+down_revision = '685cb4bc0b79'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column('bundle_store', 'storage_type', type_=sa.String(255), existing_type=sa.Enum("disk", "azure_blob"), nullable=True)
```
Comment (epicfaace) [inline alembic/versions/2021121222_rm_storage_type_enum_a325a0756797.py:19]: This is a one-time change. Realized that it's difficult to modify enums so I changed the columns to just be Text types instead, so we don't need migrations for future expansions of supporting new storage types.

**Comment by epicfaace** [comment]
@percyliang FYI this PR is about the unit of work required to support a new storage type that Apache Beam supports. Relatively straightforward and minimal -- a similar PR could add support for S3.

File: docs/Multiple-Bundle-Stores.md [hunk 3]
```diff
@@ -36,9 +36,18 @@ cl store add --name blob --storage-type azure_blob --url azfs://devstoreaccount1
 cl upload --store blob mkdocs.yml
 ```
 
-## Steps to set up GCP
+## Steps to set up and test with GCP
+
+First, make sure the `CODALAB_GOOGLE_APPLICATION_CREDENTIALS` env var is set on the server that runs the REST server. It should be set to a path to a credentials JSON file.
+
+```
+export CODALAB_GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
```
Comment (epicfaace) [inline docs/Multiple-Bundle-Stores.md:44]: @teetone Flagging this for you -- you'll need to set up this environment variable when setting up GCP on codalab.stanford.edu.

**Review by jzwang43 — APPROVED** [review]

**Pull request merged**